src/Drawer/Drawer.js

```diff
@@ -92,34 +87,244 @@ class Drawer extends React.Component {
+
+    const currentX =
+      anchor === 'right'
+        ? document.body.offsetWidth - event.touches[0].pageX
+        : event.touches[0].pageX;
+    const currentY =
+      anchor === 'bottom'
+        ? window.innerHeight - event.touches[0].clientY
+        : event.touches[0].clientY;
+
+    if (this.state.swiping) {
+      event.preventDefault();
+      this.setPosition(this.getTranslate(horizontalSwipe ? currentX : currentY));
+    } else if (this.maybeSwiping) {
+      const dXAbs = Math.abs(currentX - this.touchStartX);
+      const dYAbs = Math.abs(currentY - this.touchStartY);
+      // If the user has moved his thumb ten pixels in either direction,
+      // we can safely make an assumption about whether he was intending
+      // to swipe or scroll.
+      const threshold = 10;
+
+      const isSwiping = horizontalSwipe
+        ? dXAbs > threshold && dYAbs <= threshold
+        : dYAbs > threshold && dXAbs <= threshold;
+
+      if (isSwiping) {
+        this.swipeStartX = currentX;
+        this.swipeStartY = currentY;
+        this.setState({
+          swiping: this.props.open ? 'closing' : 'opening',
+        });
+        this.setPosition(this.getTranslate(horizontalSwipe ? currentX : currentY));
+      } else if (
+        horizontalSwipe
+          ? dXAbs <= threshold && dYAbs > threshold
+          : dYAbs <= threshold && dXAbs > threshold
+      ) {
+        this.onBodyTouchEnd();
+      }
+    }
+  };
+
+  onBodyTouchEnd = event => {
+    // sometimes the event handler is called after unbinding it
+    if (!this.mounted) return;
+
+    if (this.state.swiping) {
+      const anchor = this.getAnchor();
+      const currentX =
+        anchor === 'right'
+          ? document.body.offsetWidth - event.changedTouches[0].pageX
+          : event.changedTouches[0].pageX;
+      const currentY =
+        anchor === 'bottom'
+          ? window.innerHeight - event.changedTouches[0].clientY
+          : event.changedTouches[0].clientY;
+      const translateRatio = this.isHorizontalSwiping()
+        ? this.getTranslate(currentX) / this.getMaxTranslate()
+        : this.getTranslate(currentY) / this.getMaxTranslate();
+
+      this.maybeSwiping = false;
+      const swiping = this.state.swiping;
+      this.setState({
+        swiping: null,
+      });
+
+      // We have to open or close after setting swiping to null,
+      // because only then CSS transition is enabled.
+      if (translateRatio > 0.5) {
+        if (swiping === 'opening') {
+          this.setPosition(this.getMaxTranslate());
+        } else if (this.props.onClose != null) {
+          this.props.onClose();
+        }
+      } else if (swiping === 'opening') {
+        if (this.props.onOpen != null) {
+          this.props.onOpen();
+        }
+      } else {
+        this.setPosition(0);
+      }
+    } else if (this.maybeSwiping) {
+      if (!this.props.open && event != null) {
+        event.preventDefault(); // prevent ghost clicks
```

Comment:
I don't think we need to preventDefault on the transition end event. I was able to get along with it on the touchStart / touchMove callback.